eth/celo_state_accessor.go

```diff
@@ -0,0 +1,35 @@
+package eth
+
+import (
+	"fmt"
+
+	"github.com/celo-org/celo-blockchain/core"
+	"github.com/celo-org/celo-blockchain/core/state"
+	"github.com/celo-org/celo-blockchain/core/types"
+)
+
+// Wraps `stateAtBlock` with the additional Celo-specific parameter afterNextRandomCommit.
+// This parameter executes the random commitment at the start of the next block,
+// since this is necessary for properly tracing transactions in the next block.
+func (eth *Ethereum) celoStateAtBlock(block *types.Block, reexec uint64, base *state.StateDB, checkLive bool, preferDisk bool, afterNextRandomCommit bool) (statedb *state.StateDB, err error) {
```

Comment:
`afterNextRandomCommit` sounds a little confusing, maybe we could rename it to match the other bool arguments in the signature and sound more directive like `commitRandomness` or something similar?